test/include/bcs/ArrayHFEMDirichletTestBC.h

```diff
@@ -0,0 +1,33 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+#include "ArrayLowerDIntegratedBC.h"
+
+class ArrayHFEMDirichletTestBC : public ArrayLowerDIntegratedBC
+{
+public:
+  static InputParameters validParams();
+
+  ArrayHFEMDirichletTestBC(const InputParameters & parameters);
+
+protected:
+  virtual void computeQpResidual(RealEigenVector & residual) override;
+  virtual void computeLowerDQpResidual(RealEigenVector & residual) override;
+  virtual RealEigenVector computeLowerDQpJacobian(Moose::ConstraintJacobianType type) override;
+  virtual RealEigenMatrix computeLowerDQpOffDiagJacobian(Moose::ConstraintJacobianType type,
+                                                         const MooseVariableFEBase & jvar) override;
```

Comment:
`virtual` is redundant in the presence of `override` but we have this all throughout MOOSE so I don't ask people to change it